### PR TITLE
Fix serialization of pinned and vetoed fields

### PR DESCRIPTION
**Pull request by robfletcher**
Fixes serialization of fields in artifact/environment summary. Instead of boolean fields and objects now there are just the objects.

### CODE REVIEW

File: keel-core/src/test/kotlin/com/netflix/spinnaker/keel/serialization/BooleanPropertySerializationTests.kt [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import dev.minutest.junit.JUnit5Minutests
+import dev.minutest.rootContext
+import java.time.Instant.now
+import strikt.api.expectThat
+import strikt.jackson.at
+import strikt.jackson.has
+import strikt.jackson.isObject
+import strikt.jackson.path
+
+class BooleanPropertySerializationTests : JUnit5Minutests {
```
Comment (gal-yardeni) [inline keel-core/src/test/kotlin/com/netflix/spinnaker/keel/serialization/BooleanPropertySerializationTests.kt:15]: Should this be renamed to something else? `Boolean` with no boolean is a bit confusing :)

File: keel-core-test/src/main/kotlin/com/netflix/spinnaker/keel/persistence/ArtifactRepositoryTests.kt [hunk 2]
```diff
@@ -610,7 +610,6 @@ abstract class ArtifactRepositoryTests<T : ArtifactRepository> : JUnit5Minutests
             )
             expect {
               that(envArtifactSummary).isNotNull()
-              that(envArtifactSummary?.isPinned).isTrue()
```
Comment (gal-yardeni) [inline keel-core-test/src/main/kotlin/com/netflix/spinnaker/keel/persistence/ArtifactRepositoryTests.kt:613]: In order to keep the same functionality, should we add here (and below, if I understand the change correctly):
` that(envArtifactSummary?.pinned).isNotNull()` ?

File: keel-core/src/test/kotlin/com/netflix/spinnaker/keel/serialization/BooleanPropertySerializationTests.kt (hunk 1 above)
Comment (robfletcher) [reply keel-core/src/test/kotlin/com/netflix/spinnaker/keel/serialization/BooleanPropertySerializationTests.kt:15]: Yes, I'll do that. Good point.

File: keel-core-test/src/main/kotlin/com/netflix/spinnaker/keel/persistence/ArtifactRepositoryTests.kt (hunk 2 above)
Comment (robfletcher) [reply keel-core-test/src/main/kotlin/com/netflix/spinnaker/keel/persistence/ArtifactRepositoryTests.kt:613]: The next line does an `isEqualTo` on the same field so testing it's not null explicitly doesn't really add anything.

**Comment by robfletcher** [comment]
There was a change made in a recent version of Jackson so that those field definitions conflict. It started serializing `pinned` as `isPinned` and because there was already an `isPinned` property, one overrode the other. 

The change in behavior seems to have be prompted by this issue: https://github.com/FasterXML/jackson-module-kotlin/issues/80 unfortunately the "fix" was a breaking change. The only way I could successfully get _both_ properties to serialize was to annotate each _twice_ with `@JsonProperty` (once on the property itself, once on the getter, i.e. using `@get:JsonProperty` syntax). Anything less that 2 annotations on each property either results in one property overriding the other in the serialized output, or Jackson throwing an exception because it detects the ambiguity.

**Comment by lorin** [comment]
> The only way I could successfully get _both_ properties to serialize was to annotate each _twice_ with `@JsonProperty` (once on the property itself, once on the getter, i.e. using `@get:JsonProperty` syntax). Anything less that 2 annotations on each property either results in one property overriding the other in the serialized output, or Jackson throwing an exception because it detects the ambiguity.

🤮

**Comment by robfletcher** [comment]
@erikmunson does this look like the format that will un-fuck the environments view?

**Comment by erikmunson** [comment]
Yep looks right to me - thanks for getting it resolved! I'm hopeful there is a way to prevent backend folks from having to deal with this property overriding business in the future 🤞

**Comment by robfletcher** [comment]
> Yep looks right to me - thanks for getting it resolved! I'm hopeful there is a way to prevent backend folks from having to deal with this property overriding business in the future 🤞

Yeah, I think I'm going to raise an issue on the jackson-module-kotlin repo

**Pull request merged**